[8.11] CSPM onboarding for Azure (subscription-level) and for GCP (organization-level) (#4111)
elastic/security-docs: docs/cloud-native-security/cspm-get-started-azure.asciidoc

[[cspm-get-started-azure]]
= Get started with CSPM for Azure

[discrete]
[[cspm-overview-azure]]
== Overview

This page explains how to get started monitoring the security posture of your cloud assets using the Cloud Security Posture Management (CSPM) feature.

.Requirements
[sidebar]
--
* The CSPM integration is available to all {ecloud} users. On-premise deployments require an https://www.elastic.co/pricing[Enterprise subscription].
* To view posture data, you need `read` privileges for the following {es} indices:
** `logs-cloud_security_posture.findings_latest-*`
** `logs-cloud_security_posture.scores-*`
** `logs-cloud_security_posture.findings`

> * The user who gives the CSPM integration GCP permissions must be an Azure subscription `admin`.

Reviewer: ```suggestion
* The user who gives the CSPM integration permissions must be an Azure subscription `admin`.
```

Reviewer: Also, is this note needed here as well, given that there is another equivalent note below?

[discrete]
[[cspm-add-and-name-integration-azure]]
=== Add your CSPM integration
. From the Elastic Security *Get started* page, click *Add integrations*.
. Search for `CSPM`, then click on the result.
. Click *Add Cloud Security Posture Management (CSPM)*.
. Under **Configure integration**, select **Azure**.
. Give your integration a name that matches the purpose or team of the Azure subscription you want to monitor, for example, `azure-dev-policy`.

[discrete]
[[cspm-set-up-cloud-access-section-azure]]
=== Set up cloud account access

NOTE: To setup CSPM for an Azure subscription, you will need admin privileges for that subscription.

For most users, the simplest option is to use an Azure Resource Manager (ARM) template to automatically provision the necessary resources and permissions in Azure. If you prefer a more hands-on approach or require a specific configuration not supported by the ARM template, you can use one of manual setup options described below.

[discrete]
[[cspm-set-up-ARM]]
== ARM template setup (recommended)

. Under *Setup Access*, select *ARM Template*.
. Under **Where to add this integration**:

> .. Select **New Hosts**.

Reviewer: (nitpick, I know this is copied in all equivalent sections of all cloud providers) do we need to instruct the users to select `New Hosts`? It's the default option. We can remove the extra step to simplify instructions.

.. Name the {agent} policy. Use a name that matches the purpose or team of the cloud account or accounts you want to monitor. For example, `azure-dev-policy`. Click **Save and continue**. The *ARM Template deployment* window appears.
.. In a new tab, log in to the Azure portal, then return to {kib} and click **Launch ARM Template**. This will open the ARM template in Azure.

> .. Copy the `Fleet URL` and `Enrollment Token` that appear in {kib} to the corresponding fields in the ARM Template, then click *Review + create*.

Reviewer: (optional) We can mention that the user can optionally change the `Resource Group Name` parameter otherwise the name of the resource group is going to be `cloudbeat-` and the current epoch timestamp

. Return to {kib} and wait for the confirmation of data received from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-set-up-manual-azure]]
== Manual setup

> For manual setup, there are two authentication methods: using managed identities (recommended), or using environment variables with authentication secrets. Both methods involve deploying {agent} to a VM in the Azure subscription you want to monitor with CSPM.

Reviewer: ```suggestion
For manual setup, there are two authentication methods: using managed identities (recommended), or using environment variables with authentication secrets.
```

The benefit of the second method is that {agent} can be deployed **anywhere**, including a machine outside Azure or even your personal laptop.

[discrete]
[[cspm-azure-managed-identity-setup]]
=== Option 1: Managed identity (recommended)

This method involves creating an Azure VM (or using an existing one), giving it read access to the subscription you want to monitor with CSPM, and installing {agent} on it.

. Go to the Azure portal to https://portal.azure.com/#create/Microsoft.VirtualMachine-ARM[create a new Azure VM].
. Follow the setup process, and make sure you enable **System assigned managed identity** under the **Management** tab.
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)**, and select **Add Role Assignment**.

> . Select the `Reader` function role, assign access to `Managed Identity`, then select your VM.

Reviewer: ```suggestion
. Select the `Reader` function role, assign access to **Managed Identity**, then select your VM.
```

. Click **Save and continue**, then follow the instructions to install {agent} on your Azure VM.

Wait for the confirmation that {kib} received data from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-azure-environment-variables-setup]]
=== Option 2: Environment variables with authentication secrets

Before using this method, you must have https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in[set up a Microsoft Entra application and service principal that can access resources]. 

. Go to the **Registered apps** section of https://ms.portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps[Microsoft Entra ID].
. Click on **New Registration**, name your app and click **Register**.
. Copy your new app's `Directory (tenant) ID` and `Application (client) ID`.
. Select **Certificates & secrets**, then select **New client secret**. Copy the new secret.
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)** and select **Add Role Assignment**. 

> . Select the `Reader` function role, assign access to `User, group, or service principal`, and select your new app.

Reviewer: ```suggestion
. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.
```

On the VM where you plan to install {agent}, create the file `/etc/sysconfig/elastic-agent`. Paste the following content into the new file, substituting the values you copied for the placeholder values:

```
AZURE_TENANT_ID=<Directory (tenant) ID>
AZURE_CLIENT_ID=<Application (client) ID>
AZURE_CLIENT_SECRET=<Secret Value>
```

> After assigning the role:

Reviewer: ```suggestion
After creating the file:
```

[discrete]
[[cspm-add-and-name-integration-azure]]
=== Add your CSPM integration
. From the Elastic Security *Get started* page, click *Add integrations*.
. Search for `CSPM`, then click on the result.
. Click *Add Cloud Security Posture Management (CSPM)*.
. Under **Configure integration**, select **Azure**.
. Give your integration a name that matches the purpose or team of the Azure subscription you want to monitor, for example, `azure-dev-policy`.

[discrete]
[[cspm-set-up-cloud-access-section-azure]]
=== Set up cloud account access

NOTE: To set up CSPM for an Azure subscription, you will need admin privileges for that subscription.

For most users, the simplest option is to use an Azure Resource Manager (ARM) template to automatically provision the necessary resources and permissions in Azure. If you prefer a more hands-on approach or require a specific configuration not supported by the ARM template, you can use one of manual setup options described below.

[discrete]
[[cspm-set-up-ARM]]
== ARM template setup (recommended)

. Under *Setup Access*, select *ARM Template*.
. Under **Where to add this integration**:
.. Select **New Hosts**.

> .. Name the {agent} policy. Use a name that matches the purpose or team of the cloud account or accounts you want to monitor. For example, `azure-dev-policy`. Click **Save and continue**. The *ARM Template deployment* window appears.

Reviewer 2: It looks like we're recommending using the same name for both the integration policy (above in line 35) and the Agent policy, which seems like it could invite confusion for the user. Is it possible to suggest different names that make it clearer which policy does what?

.. In a new tab, log in to the Azure portal, then return to {kib} and click **Launch ARM Template**. This will open the ARM template in Azure.

> .. (Optional) Change the `Resource Group Name` parameter. Otherwise the name of the resource group defaults to `cloudbeat-` and a timestamp.

Reviewer 2: ```suggestion
.. (Optional) Change the `Resource Group Name` parameter. Otherwise the name of the resource group defaults to a timestamp prefixed with `cloudbeat-`.
```

.. Copy the `Fleet URL` and `Enrollment Token` that appear in {kib} to the corresponding fields in the ARM Template, then click *Review + create*.
. Return to {kib} and wait for the confirmation of data received from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-set-up-manual-azure]]
== Manual setup

> For manual setup, there are two authentication methods: using managed identities (recommended), or using environment variables with authentication secrets. The first method requires you to deploy {agent} to a VM in the Azure subscription you want to monitor with CSPM, the second method allows {agent} to be deployed anywhere, including a VM outside Azure or a personal laptop.

Reviewer 2: ```suggestion
For manual setup, there are two authentication methods: using managed identities (recommended), or using environment variables with authentication secrets. The first method requires you to deploy {agent} to a VM in the Azure subscription you want to monitor with CSPM; the second method allows {agent} to be deployed anywhere, including a VM outside Azure or a personal laptop.
```

. Go to the Azure portal to https://portal.azure.com/#create/Microsoft.VirtualMachine-ARM[create a new Azure VM].
. Follow the setup process, and make sure you enable **System assigned managed identity** under the **Management** tab.
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)**, and select **Add Role Assignment**.
. Select the `Reader` function role, assign access to **Managed Identity**, then select your VM.

After assigning the role:

. Return to the **Add CSPM** page in {kib}. 
. Under **Configure integration**, select **Azure**. Under **Setup access**, select **Manual**.
. Under **Where to add this integration**, select **New hosts**.
. Click **Save and continue**, then follow the instructions to install {agent} on your Azure VM.

Wait for the confirmation that {kib} received data from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-azure-environment-variables-setup]]
=== Option 2: Environment variables with authentication secrets

> Before using this method, you must have https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in[set up a Microsoft Entra application and service principal that can access resources]. 

Reviewer 2: Long hyperlinks can be awkward to read. This shortens the link and focuses it on the main part:
```suggestion
Before using this method, you must have set up a https://learn.microsoft.com/en-us/entra/identity-platform/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in[Microsoft Entra application and service principal] that can access resources. 
```

. Go to the **Registered apps** section of https://ms.portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps[Microsoft Entra ID].
. Click on **New Registration**, name your app and click **Register**.
. Copy your new app's `Directory (tenant) ID` and `Application (client) ID`.
. Select **Certificates & secrets**, then select **New client secret**. Copy the new secret.
. Go to your Azure subscription list and select the subscription you want to monitor with CSPM.
. Go to **Access control (IAM)** and select **Add Role Assignment**. 
. Select the `Reader` function role, assign access to **User, group, or service principal**, and select your new app.

> On the VM where you plan to install {agent}, create the file `/etc/sysconfig/elastic-agent`. Paste the following content into the new file, substituting the values you copied for the placeholder values:

Reviewer 2: Does this file name need an extension? Without it, it looks like a folder, not a file.

Reviewer: The filename is correct, linux configuration files very often do not have extensions.

Reviewer 2: Thanks for confirming, @orestisfl! I'll resolve this comment then.
